Add a not-expired stream

mcorbin opened the pull request:
Riemann has an `expired` stream, but not the opposite.

aphyr commented:
Maybe preferable to use `(where (expired? event))` and `(where (not (expired? event)) ...)`.

mcorbin commented:
`(where (expired? event))` will check the event `:state` and the event time, whereas the `expired` stream (and the `not-expired` one in my PR) only check the `:state`.

For example, with this example from the howto:

```clojure
(streams
  ; Email any expired event
  (expired (email "ops@foo.com"))

  ; Process only events which are *not* expired
  (where (not (expired? event))
    ...))
```
Events with `:state` != "expired" but with expired time will never be processed. It's rather confusing.

aphyr commented:
Yeah, we should probably update the `expired` stream to use `expired?` as well. It's a bit of a holdover from an earlier time.

mcorbin commented:
I updated the `expired` stream to use `expired?`. I also updated the `not-expired` stream. 
Filtering expired events is a common use case, so i think we should keep the `expired` stream and not deprecate it.

aphyr commented on src/riemann/streams.clj:
```diff
@@ -1295,9 +1295,18 @@
+  "Passes on expired events."
   [& children]
-  (apply match :state "expired" children))
+  (fn [event]
```
Inner functions should have names for profiling; the convention here is `(fn stream [event])`.

mcorbin replied:
I didn't know that, thanks you ;) updated.

jamtur01 commented:
LGTM.
